sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -220,21 +219,20 @@ def action_accept(self):
             ):
                 continue
             if wiz_line.sale_line_id:
-                index = self.order_id.order_line.ids.index(wiz_line.sale_line_id.id)
                 if wiz_line.units_included or force_zero_units_included:
-                    with order_form.order_line.edit(index) as line_form:
-                        wiz_line._prepare_update_so_line(line_form)
+                    wiz_line.sale_line_id.update(
+                        wiz_line._prepare_update_so_line_vals()
+                    )
                 else:
-                    to_remove.append(index)
+                    to_remove |= wiz_line.sale_line_id
             else:
                 sequence += 1
-                with order_form.order_line.new() as line_form:
-                    wiz_line._prepare_new_so_line(line_form, sequence)
-        # Remove at the end and in reverse order for not having problems
-        to_remove.reverse()
```

Comment:
You are removing this part that is important.

Reply:
The reverse is still important, and don't know why are you doing `to_remove[::-1]`